libraries/apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo/cache/normalized/ClientCacheExtensions.kt

```diff
@@ -119,20 +119,51 @@ fun ApolloClient.Builder.logCacheMisses(
   return addInterceptor(CacheMissLoggingInterceptor(log))
 }
 
-fun ApolloClient.Builder.store(store: ApolloStore, writeToCacheAsynchronously: Boolean = false): ApolloClient.Builder {
-  check(interceptors.none { it is AutoPersistedQueryInterceptor }) {
-    "Apollo: the normalized cache must be configured before the auto persisted queries"
+private class DefaultInterceptorChain(
+    private val interceptors: List<ApolloInterceptor>,
+    private val index: Int,
+) : ApolloInterceptorChain {
+
+  override fun <D : Operation.Data> proceed(request: ApolloRequest<D>): Flow<ApolloResponse<D>> {
+    check(index < interceptors.size)
+    return interceptors[index].intercept(
+        request,
+        DefaultInterceptorChain(
+            interceptors = interceptors,
+            index = index + 1,
+        )
+    )
   }
-  // Removing existing interceptors added for configuring an [ApolloStore].
-  // If a builder is reused from an existing client using `newBuilder()` and we try to configure a new `store()` on it, we first need to
-  // remove the old interceptors.
-  val storeInterceptors = interceptors.filterIsInstance<ApolloStoreInterceptor>()
-  storeInterceptors.forEach {
-    removeInterceptor(it)
```

Comment:
This is not an issue anymore!

Reply:
Good call! → https://github.com/apollographql/apollo-kotlin/pull/6612